core/types/bloom9.go

```diff
@@ -92,45 +91,60 @@ func (b *Bloom) UnmarshalText(input []byte) error {
 }
 
 func CreateBloom(receipts Receipts) Bloom {
-	bin := new(big.Int)
+	bin := make([]byte, BloomByteLength)
 	for _, receipt := range receipts {
-		bin.Or(bin, LogsBloom(receipt.Logs))
+		bin = or(bin, LogsBloom(receipt.Logs))
 	}
 
-	return BytesToBloom(bin.Bytes())
+	return BytesToBloom(bin)
 }
 
-func LogsBloom(logs []*Log) *big.Int {
-	bin := new(big.Int)
+func LogsBloom(logs []*Log) []byte {
+	bin := make([]byte, BloomByteLength)
 	for _, log := range logs {
-		bin.Or(bin, bloom9(log.Address.Bytes()))
+		bin = or(bin, bloom9(log.Address.Bytes()))
 		for _, b := range log.Topics {
-			bin.Or(bin, bloom9(b[:]))
+			bin = or(bin, bloom9(b[:]))
 		}
 	}
 
 	return bin
 }
 
-func bloom9(b []byte) *big.Int {
-	b = crypto.Keccak256(b)
-
-	r := new(big.Int)
+func bloom9(data []byte) []byte {
+	hash := make([]byte, 32)
```

Comment:
Also a funny thing: we're only using `6` bytes of the hash. So it's sufficient to do use a 6-byte buffer here